services/components/containers/resources.go

```diff
@@ -0,0 +1,71 @@
+package containers
+
+import (
+	"context"
+	"time"
+
+	"github.com/forta-network/forta-core-go/domain"
+	"github.com/forta-network/forta-core-go/protocol"
+	"github.com/forta-network/forta-node/clients"
+	"github.com/forta-network/forta-node/clients/docker"
+	"github.com/forta-network/forta-node/services/components/metrics"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	defaultPollingInterval = time.Second * 10
+)
+
+// PollDockerResources gets CPU and MEM usage for all agent containers with docker stats.
+func PollDockerResources(
+	ctx context.Context, dockerClient clients.DockerClient,
+	msgClient clients.MessageClient,
+) {
+	for {
+		select {
+		case <-ctx.Done():
+			logrus.Info("stopping docker resources poller")
+			return
+		case <-time.After(defaultPollingInterval):
```

Comment:
If we define a `time.Ticker` before the `for` statement and listen to the channel of that here, the timing could be more airtight. `time.After` starts counting after the handling is done and this might create drift in the start/end seconds. Also, starting this cycle crisply in the beginning of a minute (or 10 second interval in a minute) could help in aligning the timing with other instances of the bot running on different nodes. Then we could see more similar stats for a bot across the nodes.